docs/content/guides/dagster/building-versioned-assets.mdx

```diff
@@ -0,0 +1,291 @@
+---
+title: Building and memoizing versioned assets | Dagster Docs
+---
+
+# Building and memoizing versioned assets
+
+Thie guide shows you how to build memoizable graphs of assets. Using this style, one can avoid unnecessary recomputations, speeding up the developer workflow and saving computational resources.
+
+---
+
+# Context
+
+Dagster embraces functional programming principles. The core abstraction is a software-defined asset which, in its idealized form, is a side-effect-free function that produces an asset, with upstream assets as inputs.
```

Comment:
We've gotten a body of feedback that style of communication we use in our docs goes over the heads of people who don't have a rigorous CS background.

IMO we should see if communicate this in a way that assumes less knowledge / prior understanding on behalf of the reader.

cc @smackesey who's taking this over